src/themes/material/components/edit-chip-list/edit-chip-list.component.ts

```diff
@@ -0,0 +1,198 @@
+import {
+	Component,
+	Input,
+	OnInit,
+	ChangeDetectorRef,
+	ViewChildren,
+	QueryList
+} from '@angular/core';
+
+import {AppError} from 'ng2-qgrid/core/infrastructure';
+import {Command, CommandManager} from 'ng2-qgrid/core/command';
+import {Shortcut, ShortcutManager} from 'ng2-qgrid/core/shortcut';
+import {EditChipComponent} from './edit-chip/edit-chip.component';
+import {ChipState} from './edit-chip/edit-chip.component';
+
+interface ISelected {
+	index?: number;
+	value?: string;
+	action: Actions;
+}
+
+interface IChipItem {
+	value?: string | any;
+	state?: string;
+	qGridFocusEnabled?: boolean;
+}
+
+enum Actions {
+	add,
```

Comment:
explain please why u need this?

Reply:
Yes, it might be this enum is not needed in this component. But it seems that the Command pattern (where I used the enum values to decide what the action expected at the moment) is not needed in this case as well?
So I can refactor the code, to get rid both of this enum and unnecessary Command pattern using. How's that?

Reply:
Removed this enum, (as well as NextAction enum from chip.component.ts)
(commit: [Merge from Dev: conflicts resolving](https://github.com/qgrid/ng2/pull/39/commits/da05f15a228a611a5059e3049c9f18402446fa8f)).